### ISSUE TITLE
Introduce sample duplicate detection or show pending samples [2]

### ISSUE THREAD

**Issue by kwa20** [hzi-braunschweig/SORMAS-Project#4283]
### Situation Description
Thus far, it is not possible to only document pathogen test results to already existing samples for cases through the DEMIS interface. Every time, new samples will be created even if users have already deposited collected samples in Sormas. This is often the case when health departments perform the sample collection and subsequently send the sample to the laboratories. It is therefore needed to merely add a test result to an existing sample through the interface.
### Feature Description
During the lab message report processing, introduce a sample duplicate check for users to be able to choose existing samples for which test results can then be added. This will prevent the creation of duplicate samples for respective cases. 
### Possible Alternatives
As an alternative, show the user all pending samples when they have gone through person and case duplicate detection already and let them either select an existing sample for which the result was sent via DEMIS or give them the option to create a new sample instead. This might be an easier solution since the number of pending samples is expected to be low and data coming from DEMIS is not always reliable.

### Additional Information
1. check: use LabSampleID if available 
2. check:  date plus (+-2 days) sample material for duplicate detection

**Comment by bernardsilenou** [comment]
@kwa20  This need is similar to #1154, Please read the comments on Jan 22, 2021 from @Iheanacho2027
How do you the duplicate detection should work? variables to be used?
Points to consider:
The main challenge is that by the time the user has a sample, there are 3 possibilities:
 1. Case may have been created in SORMAS and more then one samples documented, so we  need a duplicate mechanism here to identify the right case + right sample among the samples of the case+ if the test result already exist.
 2. Case may have been created but no sample was document, here we need a mechanism to identify the source case, create sample + pathogen result
 3. Case may not have been created, her we need a duplicate mechanism to validate that case does not exist + creating one+ sample+pathogen test

In summary, we would need duplicate detection for: case, sample, and pathogen test  and then integrate all in a method.

**Comment by leventegal-she** [comment]
In the current flow all samples of the selected case/contact/event participant is displayed so the user can choose one or create a new one.

Currently the sample date and material are not mapped from climedo so this kind of duplicate check doesn't find any duplicates.

It would be better to postpone the implementation of this feature until those fields will be mapped.
@MateStrysewske @markusmann-vg  what do you think?

**Comment by MateStrysewske** [comment]
@leventegal-she I think that's perfectly fine.

**Comment by AndyBakcsy-she** [comment]
Validated ticket on the latest version of sormas 1.58.0-snapshot
